circuits/test/PointInPolygon.js

```diff
@@ -0,0 +1,949 @@
+const chai = require("chai");
+const chaiAsPromised = require("chai-as-promised");
+chai.use(chaiAsPromised);
+chai.should();
+
+const path = require("path");
+
+const wasm_tester = require("circom_tester").wasm;
+
+const F1Field = require("ffjavascript").F1Field;
+const Scalar = require("ffjavascript").Scalar;
+exports.p = Scalar.fromString(
+  "21888242871839275222246405745257275088548364400416034343698204186575808495617"
+);
+const Fr = new F1Field(exports.p);
+
+const assert = chai.assert;
+
+describe("Simple Polygon", function () {
+  this.timeout(100000000);
+
+  let circuit;
+  this.beforeAll(async () => {
+    const filepath = path.join(__dirname, "Simple5.circom");
+    circuit = await wasm_tester(filepath);
+    await circuit.loadConstraints();
+    assert.equal(circuit.constraints.length, 22009); // TODO: verify that this is expected
+  });
+
+  const test_permutations = async (polygon, result) => {
+    // cycle points around
+    i = 0;
+    while (i < 5) {
+      i++;
+      polygon.unshift(polygon.pop());
+      var witness = await circuit.calculateWitness({ polygon: polygon }, true);
+      assert(Fr.eq(Fr.e(witness[1]), Fr.e(result)));
+
+      // try the reversed polygon
+      polygon.reverse();
+      var witness = await circuit.calculateWitness({ polygon: polygon }, true);
+      assert(Fr.eq(Fr.e(witness[1]), Fr.e(result)));
+      polygon.reverse();
+    }
+  };
+
+  it("Should make sure we can't repeat points in sequence", async () => {
+    await chai
+      .expect(
+        test_permutations(
+          [
+            [0, 0],
+            [0, 0],
+            [5, 0],
+            [5, 5],
+            [0, 5],
+          ],
+          0
+        ) // shape is a counterclockwise square
+      )
+      .to.eventually.be.rejectedWith("Assert Failed"); // can't do line intersections for lines that are just points
```

Comment:
There are two stylistic changes here.
1: Using a variable rather than passing it straight to the the function.
2: Defining the polygons on one line.

I agree about 1, but I kind of like having each point on a different line, I feel like it's easier to read. Is this from prettier, or shall we get @0xPetra to do the tie break?